.github/workflows/check-public-api.yaml

```diff
@@ -0,0 +1,52 @@
+name: Check public API
+
+env:
+  RUST_BACKTRACE: 1
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]   
```

Comment:
We don't want this to break on every commit, better remove the on push and on pull_request events and put workflow_dispatch (that's how github calls manual actions)

Reply:
Yes, this is a draft PR, this part will be removed later on, I just needed to trigger it once in the PR.

Also I need it to be `master`, not `main`